### PR TITLE
Upgrade eslint plugin vue

### PR DESCRIPTION
**Pull request by MisRob**
### Summary

- upgrade eslint-plugin-vue to the latest version
- manually fix issues regarding a new rule vue/return-in-computed-property
- run auto fix

### Reviewer guidance

- run lint
- are new return statements OK?

### References

Solves: first part of #4088 
Follow-ups: #5337, #5338

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

**Comment by indirectlylit** [comment]
cool. were these changes made automatically, or did you need to manually update them to get rid of the linting errors?

**Comment by MisRob** [comment]
It was both - you can find manual fixes for the new rule in the second commit and auto fixes in the third, last, commit.

File: kolibri/plugins/device_management/assets/src/views/SelectContentPage/index.vue [hunk 1]
```diff
@@ -185,6 +185,8 @@
         if (Object.values(ContentWizardErrors).includes(this.status)) {
           return this.status;
         }
+
+        return undefined;
```
Comment (jonboiser) [inline kolibri/plugins/device_management/assets/src/views/SelectContentPage/index.vue:189]: Have this return the same type as `this.status`. Is it a string?

File: kolibri/plugins/learn/assets/src/views/ContentPage.vue [hunk 2]
```diff
@@ -230,6 +234,8 @@
         if (breadcrumbs.length > 0) {
           return breadcrumbs[breadcrumbs.length - 1];
         }
+
+        return undefined;
```
Comment (jonboiser) [inline kolibri/plugins/learn/assets/src/views/ContentPage.vue:238]: Have this return `null`, so the return type is `Maybe (breadcrumb)`, JS-style.

**Comment by jonboiser** [comment]
This looks good. IT seems the only mandatory new rules from this upgrade are the "Newline after tags" rule and "required return statement in functions" rule.

For the "required return statement" rule, I think the conservative thing that was done was to return `undefined` to make the  function semantically equivalent, but we should probably take a look at some of these spaces and see if we can have a little more discipline on return types. For example, always return a number, or an optional type (e.g. number or null).

**Comment by rtibbles** [comment]
> For the "required return statement" rule, I think the conservative thing that was done was to return undefined to make the function semantically equivalent, but we should probably take a look at some of these spaces and see if we can have a little more discipline on return types. For example, always return a number, or an optional type (e.g. number or null).

I agree with this - if we want to be disciplined, we could require a single type to return (in the case of a `Number` we should use `NaN` instead of `null` to enforce this). The edge case that comes to mind is if there are any instances where we might want to distinguish between `''` and a truly `undefined` value.

**Comment by indirectlylit** [comment]
I think `null` is an appropriate placeholder for combining with any return type when you're trying to say "things are working fine, but here it doesn't exist", similar to `None` in Python.

Usually if I see `undefined` or `NaN` I assume something in the code has malfunctioned, and `''` is of course a valid string.

**Comment by MisRob** [comment]
Yes, for now, I stayed with `undefined` in those properties where I wasn't (at least 99% ) sure that I won't break anything by using more specific type because, well, it's JavaScript and I've seen too much I guess :))

I agree that it would be nice to follow some standards when it comes to return types. Using `null` as expected none value for any type makes sense to me, too. I also see `undefined` or `NaN` rather as something unexpected.

I'm going to open an issue and reference this discussion there.

**Review by jonboiser — APPROVED** [review]

**Pull request merged**